### PR TITLE
Add feature: Only inline when matched once

### PR DESCRIPTION
**Pull request by strarsis**
This PR will add the option to only inline CSS when it matches only once.

### CODE REVIEW

File: lib/inline.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 'use strict';
 
-var utils = require('./utils');
+var utils  = require('./utils'),
+    mensch = require('mensch'),
+    equal  = require('deep-equal');
```
Comment (jrit) [inline lib/inline.js]: `var` on each line with semicolon at end please for all declarations.

File: lib/inline.js [hunk 2]
```diff
@@ -318,6 +333,60 @@ function juiceDocument($, options) {
...
+    if (styleDataList.length !== 1) {
+      return;
+    }
+
+    var ast      = mensch.parse(styleElement.childNodes[0].nodeValue, {position: true, comments: true});
+    var rulesAst = ast.stylesheet.rules;
+
+    var changed  = false;
+    for(var ruleAst of rulesAst) {
+loop2:
```
Comment (jrit) [inline lib/inline.js]: Seems like this `loop2:` is just hanging out here. Is this a mistake?

**Comment by jrit** [comment]
Was this strategy a compromise over the previously discussed behavior around minimizing the overall number of characters because implementing that was just too complicated?

**Comment by strarsis** [comment]
Yes, it is a compromise. In most cases it should be better to not inline when matching more than once.

**Comment by jrit** [comment]
@strarsis there is a couple nits with code style. If this change is enough to get the svgo change unblocked I'll merge after those are addressed. Ultimately it would be nice to do it based on character count, but I don't want perfection to stand in the way of progress here.

**Comment by strarsis** [comment]
@jrit: https://github.com/svg/svgo/pull/592#issuecomment-252683045

**Comment by jrit** [comment]
Interesting. I don't think it makes sense to expand juice to support node prior to v4 seeing as official support for 0.10 ends Monday and support for 0.12 ends on December 31st. Shall I leave the PR open for now or do you want to close it?

**Comment by strarsis** [comment]
I would think best would be closing this PR for now and reopen it when svgo doesn't require support for node v0.10 anymore.

**Pull request closed without merging**